File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.java
// Last updated: 5/20/2026, 11:37:50 PM
import java.util.Arrays;
import java.util.HashSet;

class Solution {
    public int removeDuplicates(int[] nums) {
       if (nums.length == 0) {
            return 0;
        }

        int uniqueCount = 1; // At least the first element is unique

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != nums[i - 1]) {
                nums[uniqueCount] = nums[i];
                uniqueCount++;
            }
        }

        return uniqueCount;
    }
}

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.cpp
// Last updated: 5/20/2026, 11:37:48 PM
#include <cstdio>
#include <set>
#include <vector>
using std::vector;

class Solution {
public:
  int removeDuplicates(vector<int> &nums) {

    for (int i = 0; i < nums.size(); ++i) {
      auto curr = nums.at(i);

      nums.erase(std::remove_if(nums.begin() + i + 1, nums.end(),
                                [&](int &val) { return val == curr; }),
                 nums.end());
    }

    return nums.size();
  }
};
